specs/CustomDownload.md

```diff
@@ -0,0 +1,528 @@
+  /// exist, it will be created.
+  [propget] HRESULT ResultFilePath([out, retval] LPWSTR* resultFilePath);
+
+  /// Sets the `ResultFilePath` property.
+  [propput] HRESULT ResultFilePath([in] LPCWSTR resultFilePath);
+
+  /// Returns an `ICoreWebView2Deferral` object.  Use this operation to
+  /// complete the event at a later time.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+}
+
+/// Represents a download process. Gives access to the download's metadata
+/// and supports a user canceling, pausing, or resuming the download.
+[uuid(0C4A07B1-B610-459F-A574-EC253D5EC40D), object, pointer_default(unique)]
+interface ICoreWebView2DownloadItem : IUnknown
+{
+  /// The unique ID of the download.
+  [propget] HRESULT Id([out, retval] UINT32* id);
+
+  /// The URI of the download.
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The Content-Disposition header value from HTTP response.
+  [propget] HRESULT ContentDisposition([out, retval] LPWSTR* contentDisposition);
+
+  /// MIME type of the downloaded content.
+  [propget] HRESULT MimeType([out, retval] LPWSTR* mimeType);
+
+  /// The size of the download in total number of expected bytes.
+  [propget] HRESULT DownloadSizeInBytes([out, retval] UINT64* downloadSizeInBytes);
+  
+  /// The number of bytes that have been written to the download file.
+  [propget] HRESULT ReceivedBytes([out, retval] UINT64* receivedBytes);
+
+  /// The estimated end time in ISO 8601 format.
+  [propget] HRESULT EstimatedEndTime([out, retval] LPWSTR* estimatedEndTime);
+
+  /// The absolute path to the file, including file name. Host can change
+  /// this from `ICoreWebView2DownloadStartingEventArgs`.
+  [propget] HRESULT ResultFilePath([out, retval] LPWSTR* resultFilePath);
+
+  /// The danger type of the download. The danger type can be none, file, content,
+  /// uncommon, host, or potentially unwanted. See `COREWEBVIEW2_DOWNLOAD_DANGER_TYPE`
+  /// for more details.
+  [propget] HRESULT DangerType([out, retval] COREWEBVIEW2_DOWNLOAD_DANGER_TYPE* dangerType);
+
+  /// The state of the download. A download can be in progress, interrupted, or
+  /// complete.
+  [propget] HRESULT State([out, retval] COREWEBVIEW2_DOWNLOAD_STATE* downloadState);
+
+  /// The reason why a download was interrupted.
+  [propget] HRESULT InterruptReason([out, retval] LPWSTR* interruptReason);
+
+  /// Cancels the download. If canceled, the default download dialog will show that the
+  /// download was canceled. Host should set `Cancel` from `ICoreWebView2SDownloadStartingEventArgs`
+  /// if cancellation should be hidden from default download dialog.
+  HRESULT Cancel();
+
+  /// Pauses the download. If paused, the default download dialog will
+  /// show that the download is paused. No effect if download is already paused.
+  HRESULT Pause();
+
+  /// Resumes a paused download. No effect if download is not paused.
+  HRESULT Resume();
+}
```

Comment:
Don't we need an event on the Download class to say when the state changes or the receivedbytes and so on properties change?